### ISSUE TITLE
How to get predicted target point from result and source point

### ISSUE THREAD

**Issue by binhmuc** [ignacio-rocco/cnngeometric_pytorch#11]
Thank for your source code !
Could  you tell me how to get target point from pretrained model and the source point.
I looking for "eval_pf.py" but look like you get source point from target point...

**Comment by binhmuc** [comment]
Also your paper said that "A keypoint is considered to be matched correctly if its predicted location is within a distance of α · max(h, w) of the target keypoint position". So i don't know why you code compare with source points instead of target points.

**Comment by ignacio-rocco** [comment]
Hi, this is just a matter of terminology. For consistency with the
ProposalFlow paper that uses inverse warping we also transform the points
from the second image (target) to the first image (source). In the paper we
explain it in the opposite way because it's more natural. But remember
"source" and "target" are just two names. Sorry for the confusion.

**Comment by binhmuc** [comment]
Thank for you reply :)
So, it means that, i just replace "source points" and "target point" in the code and got the natural result ?.
But it too weird for me...
Because in the code: You warped source images -> target images, and using theta result to get inverse warping...
So, could you tell me, how to get target point from source points and theta result ?
Thank you !

**Comment by binhmuc** [comment]
So, maybe i will change the format of input size which feed to the model: inverse source images and target images. After that,  replace "source points" and "target point" concept to get natural result ?
Thank you.

**Comment by ignacio-rocco** [comment]
Please see the explanations about inverse warping here:

https://www.cs.unc.edu/~lazebnik/research/fall08/lec08_faces.pdf

this should help you understand!

**Comment by lixiaolusunshine** [comment]
so  do you understand his means? I'm also confused this opinions.

**Comment by binhmuc** [comment]
@lixiaolusunshine  yes, i understood him. Clearly that, the paper said that source points to target points, but in the source code is totally inverse.

**Comment by lixiaolusunshine** [comment]
so in his paper he got the estimated inverse affine parameters from the featuregression layer, then use this inverse mapping to warp the source image into the target image?

**Comment by binhmuc** [comment]
@lixiaolusunshine  sorry, i cannot catch up your mean. In his paper, very clear that, use GMM, find a list of parameters, from parameters => warp => loss. 
The only difference is when he compare the result. He compare the target points, but in code, we never get target points for the parameters, instead of is source points.

**Comment by tkrtr** [comment]
@binhmuc 
Thanks for your issue.

Link about inverse points method is broken.
If you know how to do the inverse points, I would like to know.
The owner of this source code does not appear to be replying at this time.

> Please see the explanations about inverse warping here:
> 
> https://www.cs.unc.edu/~lazebnik/research/fall08/lec08_faces.pdf
> 
> this should help you understand!
